From b77d66680283670ccfb7277c61f0360752f04be1 Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Fri, 8 Dec 2023 20:23:02 +0100
Subject: [PATCH 03/27] parisc/pciinit: Show if a PCI device is 64-bit device
Forwarded: yes

Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/fw/pciinit.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/roms/seabios-hppa/src/fw/pciinit.c b/roms/seabios-hppa/src/fw/pciinit.c
index 55f331a7..1535fa4d 100644
--- a/roms/seabios-hppa/src/fw/pciinit.c
+++ b/roms/seabios-hppa/src/fw/pciinit.c
@@ -1148,9 +1148,10 @@ pci_region_map_one_entry(struct pci_region_entry *entry, u64 addr)
 {
     if (entry->bar >= 0) {
         dprintf(1, "PCI: map device bdf=%pP"
-                "  bar %d, addr %08llx, size %08llx [%d: %s]\n",
+                "  bar %d, addr %08llx, size %08llx [%d: %s], 64bit:%d\n",
                 entry->dev,
-                entry->bar, addr, entry->size, entry->type, region_type_name[entry->type]);
+                entry->bar, addr, entry->size, entry->type,
+                region_type_name[entry->type], entry->is64);
 
         pci_set_io_region_addr(entry->dev, entry->bar, addr, entry->is64);
         return;
-- 
2.39.2

