From 426ed23a9bee9ad12b47b3129e135fe0a5685a04 Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Wed, 13 Dec 2023 20:57:51 +0100
Subject: [PATCH 06/23] parisc/scsi: Set default max block size if not set by
 driver

Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/parisc/parisc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/roms/seabios-hppa/src/parisc/parisc.c b/roms/seabios-hppa/src/parisc/parisc.c
index 4618f0f7..c223161c 100644
--- a/roms/seabios-hppa/src/parisc/parisc.c
+++ b/roms/seabios-hppa/src/parisc/parisc.c
@@ -1178,7 +1178,10 @@ int __VISIBLE parisc_iodc_ENTRY_IO(unsigned int *arg FUNC_MANY_ARGS)
 
                 // Make sure we know how many bytes we can read at once!
                 // NOTE: LSI SCSI can not read more than 8191 blocks, esp only 64k
-                BUG_ON(disk_op.drive_fl->max_bytes_transfer == 0);
+                if (disk_op.drive_fl->max_bytes_transfer == 0) {
+                    dprintf(0, "WARNING: Maximum transfer size not set for boot disc.\n");
+                    disk_op.drive_fl->max_bytes_transfer = 64*1024;   /* 64kb */
+                }
 
                 if (option == ENTRY_IO_BBLOCK_IN) { /* in 2k blocks */
                     unsigned long maxcount;
-- 
2.39.2

