From a3653ada7cc3474d0a6c4a0593b7517386921a8d Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Tue, 2 Jan 2024 23:22:19 +0100
Subject: [PATCH 13/23] parisc: Move x86-only code inside ifdef

Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/output.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/roms/seabios-hppa/src/output.c b/roms/seabios-hppa/src/output.c
index b940480e..9892ea72 100644
--- a/roms/seabios-hppa/src/output.c
+++ b/roms/seabios-hppa/src/output.c
@@ -482,6 +482,7 @@ hexdump(const void *d, int len)
 static void
 dump_regs(struct bregs *regs)
 {
+#if CONFIG_X86
     if (!regs) {
         dprintf(1, "  NULL\n");
         return;
@@ -492,6 +493,7 @@ dump_regs(struct bregs *regs)
     dprintf(1, "  si=%08x di=%08x bp=%08x sp=%08x cs=%04x ip=%04x  f=%04x\n"
             , regs->esi, regs->edi, regs->ebp, (u32)&regs[1]
             , regs->code.seg, regs->code.offset, regs->flags);
+#endif
 }
 
 // Report entry to an Interrupt Service Routine (ISR).
-- 
2.39.2

