From ec1c829f6954460fccce7632f08b48dcb0e1f5e7 Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Tue, 2 Jan 2024 23:41:24 +0100
Subject: [PATCH 14/27] parisc: Move x86-only code inside ifdef
Forwarded: yes

Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/sercon.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/roms/seabios-hppa/src/sercon.c b/roms/seabios-hppa/src/sercon.c
index 8ae1f9cf..11c0e26e 100644
--- a/roms/seabios-hppa/src/sercon.c
+++ b/roms/seabios-hppa/src/sercon.c
@@ -513,6 +513,7 @@ void sercon_setup(void)
     if (!CONFIG_SERCON)
         return;
 
+#if CONFIG_X86
     struct segoff_s seabios, vgabios;
     u16 addr;
 
@@ -542,6 +543,7 @@ void sercon_setup(void)
     SET_LOW(sercon_port, addr);
     outb(0x03, addr + SEROFF_LCR); // 8N1
     outb(0x01, addr + SEROFF_IIR); // enable fifo
+#endif
 }
 
 /****************************************************************
-- 
2.39.2

