From bc4933cc75dd144680beb42dade7f0ce93483f98 Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Tue, 2 Jan 2024 23:43:57 +0100
Subject: [PATCH 16/27] parisc: Disable warnings on 64-bit firmware build
Forwarded: yes

Signed-off-by: Helge Deller <deller@gmx.de>
---
 Makefile.parisc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/roms/seabios-hppa/Makefile.parisc b/roms/seabios-hppa/Makefile.parisc
index 4240e0cf..36edc0c2 100644
--- a/roms/seabios-hppa/Makefile.parisc
+++ b/roms/seabios-hppa/Makefile.parisc
@@ -79,6 +79,11 @@ COMMONCFLAGS += $(call cc-option,$(CC),-fno-stack-protector,)
 COMMONCFLAGS += $(call cc-option,$(CC),-fno-stack-protector-all,)
 COMMONCFLAGS += $(call cc-option,$(CC),-fstack-check=no,)
 COMMONCFLAGS += $(call cc-option,$(CC),-mfast-indirect-calls,)
+
+COMMONCFLAGS-64 = -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast
+COMMONCFLAGS-32 =
+COMMONCFLAGS    += $(COMMONCFLAGS-$(BITS))
+
 COMMA := ,
 
 CFLAGS32FLAT := $(COMMONCFLAGS) -DMODE16=0 -DMODESEGMENT=0
-- 
2.39.2

