From 6db8c1474b0cf71f1f53602349d927111b745278 Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Tue, 2 Jan 2024 23:46:14 +0100
Subject: [PATCH 17/23] ehci: Fix build on 64-bit

Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/hw/usb-ehci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/roms/seabios-hppa/src/hw/usb-ehci.c b/roms/seabios-hppa/src/hw/usb-ehci.c
index d01fa853..3eb05f1f 100644
--- a/roms/seabios-hppa/src/hw/usb-ehci.c
+++ b/roms/seabios-hppa/src/hw/usb-ehci.c
@@ -555,7 +555,7 @@ ehci_send_pipe(struct usb_pipe *p, int dir, const void *cmd
 
     // Allocate tds on stack (with required alignment)
     u8 tdsbuf[sizeof(struct ehci_qtd) * STACKQTDS + EHCI_QTD_ALIGN - 1];
-    struct ehci_qtd *tds = (void*)ALIGN((u32)tdsbuf, EHCI_QTD_ALIGN), *td = tds;
+    struct ehci_qtd *tds = (void*)ALIGN((unsigned long)tdsbuf, EHCI_QTD_ALIGN), *td = tds;
     memset(tds, 0, sizeof(*tds) * STACKQTDS);
 
     // Setup transfer descriptors
-- 
2.39.2

