From a7efc718498cdbc1d547938bfbb308e177c45a8e Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Tue, 2 Jan 2024 23:49:11 +0100
Subject: [PATCH 18/27] memmap: Fix 64-bit build
Forwarded: yes

Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/memmap.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/roms/seabios-hppa/src/memmap.h b/roms/seabios-hppa/src/memmap.h
index 22bd4bcb..358c0673 100644
--- a/roms/seabios-hppa/src/memmap.h
+++ b/roms/seabios-hppa/src/memmap.h
@@ -7,15 +7,15 @@
 #define PAGE_SIZE 4096
 #define PAGE_SHIFT 12
 
-static inline u32 virt_to_phys(void *v) {
-    return (u32)v;
+static inline unsigned long virt_to_phys(void *v) {
+    return (unsigned long)v;
 }
-static inline void *memremap(u32 addr, u32 len) {
+static inline void *memremap(unsigned long addr, u32 len) {
     return (void*)addr;
 }
 
 // Return the value of a linker script symbol (see scripts/layoutrom.py)
-#define SYMBOL(SYM) ({ extern char SYM; (u32)&SYM; })
+#define SYMBOL(SYM) ({ extern char SYM; (unsigned long)&SYM; })
 #define VSYMBOL(SYM) ((void*)SYMBOL(SYM))
 
 #endif // memmap.h
-- 
2.39.2

