From ca2a5bf30744a00a4bff39edee9d08d01350c036 Mon Sep 17 00:00:00 2001
From: Helge Deller <deller@gmx.de>
Date: Wed, 3 Jan 2024 11:59:27 +0100
Subject: [PATCH 21/27] parisc: Fix FPU detection on NetBSD
Forwarded: yes

Signed-off-by: Helge Deller <deller@gmx.de>
---
 src/parisc/parisc.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/roms/seabios-hppa/src/parisc/parisc.c b/roms/seabios-hppa/src/parisc/parisc.c
index df85f1cc..afe248a1 100644
--- a/roms/seabios-hppa/src/parisc/parisc.c
+++ b/roms/seabios-hppa/src/parisc/parisc.c
@@ -1625,12 +1625,12 @@ static int pdc_coproc(unsigned int *arg)
     switch (option) {
         case PDC_COPROC_CFG:
             memset(result, 0, 32 * sizeof(unsigned long));
-            mask = 1UL << 7;    /* bit for FPU available/functional */
+            mask = 3UL << 6;    /* bit for FPU available/functional */
             mtctl(mask, 10);    /* initialize cr10 */
-            result[0] = mask;
-            result[1] = mask;
-            result[17] = 1;     /* Revision */
-            result[18] = current_machine->pdc_cpuid >> 5; /* CPU Model */
+            result[0] = mask;   /* ccr_enable */
+            result[1] = mask;   /* ccr_present */
+            result[17] = 1;     /* FPU revision */
+            result[18] = current_machine->pdc_cpuid >> 5; /* FPU model */
             return PDC_OK;
     }
     return PDC_BAD_OPTION;
-- 
2.39.2

